torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -253,17 +253,29 @@ void append_overloaded_arg(std::vector<py::handle> &overloaded_args, PyObject* o
   }
 }
 
-auto FunctionParameter::check(PyObject* obj, std::vector<py::handle> &overloaded_args) -> bool
+static auto check_tensor_or_overload(PyObject* obj, std::vector<py::handle> &overloaded_args) -> bool {
+  if (THPVariable_CheckExact(obj)) {
+    // torch.Tensor instances (not subclasses)
+    return true;
+  }
+
+  if (check_has_torch_function(obj)) {
+    // tensor subclasses and unrelated objects with __torch_function__
+    append_overloaded_arg(overloaded_args, obj);
+    return true;
+  } else if (THPVariable_Check(obj)) {
+    // tensor subclasses without __torch_function__
+    return true;
+  }
+
+  return false;
+}
+
+  auto FunctionParameter::check(PyObject* obj, std::vector<py::handle> &overloaded_args, int argnum) -> bool
```

Comment:
incorrect indentation

Reply:
It would be pretty helpful if there were a comment here saying what `argnum` meant